documentation/docs/05-load.md

```diff
@@ -2,253 +2,282 @@
+#### params
 
-// @filename: index.js
-// ---cut---
-/** @type {import('./$types').LayoutServerLoad} */
-export async function load({ parent }) {
-	const { a } = await parent();
-	console.log(a); // `1`
+`params` is derived from `url.pathname` and `routeId`.
+
+Given a `routeId` of `a/[b]/[...c]` and a `url.pathname` of `/a/x/y/z`, the `params` object would look like this:
 
-	return { b: 2 };
+```json
+{
+	"b": "x",
+	"c": "y/z"
 }
 ```
 
-```js
-/// file: src/routes/foo/+page.server.js
-// @filename: $types.d.ts
-export type PageServerLoad = import('@sveltejs/kit').Load<{}, null, { a: number, b: number }>;
+### Making fetch requests
 
-// @filename: index.js
-// ---cut---
-/** @type {import('./$types').PageServerLoad} */
-export async function load({ parent }) {
-	const { a, b } = await parent();
-	console.log(a, b); // `1`, `2`
+To get data from an external API or a `+server.js` handler, you can use the provided `fetch` function, which behaves identically to the [native `fetch` web API](https://developer.mozilla.org/en-US/docs/Web/API/fetch) with a few additional features:
 
-	return { c: 3 };
+- it can be used to make credentialed requests on the server, as it inherits the `cookie` and `authorization` headers for the page request
+- it can make relative requests on the server (ordinarily, `fetch` requires a URL with an origin when used in a server context)
+- internal requests (e.g. for `+server.js` routes) go direct to the handler function when running on the server, without the overhead of an HTTP call
+- during server-side rendering, the response will be captured and inlined into the rendered HTML. Note that headers will _not_ be serialized, unless explicitly included via [`filterSerializedResponseHeaders`](/docs/hooks#server-hooks-handle). Then, during hydration, the response will be read from the HTML, guaranteeing consistency and preventing an additional network request - if you got a warning in your browser console when using the browser `fetch` instead of the `load` `fetch`, this is why.
+
+```js
+/// file: src/routes/items/[id]/+page.js
+/** @type {import('./$types').PageLoad} */
+export async function load({ fetch, params }) {
+	const res = await fetch(`/api/items/${params.id}`);
+	const item = await res.json();
+
+	return { item };
 }
 ```
 
-In `+page.js` or `+layout.js` it will return data from `load` functions in parent `+layout.js` files. Implicitly, a missing `+layout.js` is treated as a `({ data }) => data` function, meaning that it will also return data from parent `+layout.server.js` files.
+> Cookies will only be passed through if the target host is the same as the SvelteKit application or a more specific subdomain of it.
 
-Be careful not to introduce accidental waterfalls when using `await parent()`. If for example you only want to merge parent data into the returned output, call it _after_ fetching your other data.
+### Cookies and headers
```

Comment:
Basically that cookies _aren't_ just a special case of headers — `cookies.set(...)` causes a `set-cookie` header to be appended, but it also updates the cookie store that accompanies the `RequestEvent`. Meanwhile `cookies.get(...)` is nothing to do with `response.headers`, it reads from `request.headers` and the event's cookie store. Headers are the _mechanism_ for dealing with cookies, but I think it's helpful to treat them as somewhat conceptually different — most headers just affect the response, but `set-cookie` affects the user agent.

In general I think it's good to be able to refer back to things that were previously covered, so that you're layering knowledge. Otherwise someone reads about `setHeader`, thinks 'great, I can use this to set a cookie', _then_ learns that actually no, they have to learn this whole other thing.